std/deps.ln

```diff
@@ -1,34 +1,333 @@
-export fn commit() {
+export event install: Package
+
+fn package(): Package {
+  let emptyStd = new Array<string> [];
+  let emptyBlock = new Array<string> [];
+  let emptyFullBlock = new Array<string> [];
+  let emptyDep = new Array<Dependency> [];
+  let pkg = new Package {
+    std: emptyStd,
+    dependencies: emptyDep,
+    block: emptyBlock,
+    fullBlock: emptyFullBlock,
+  };
+  return pkg;
+}
+
+export fn using(pkg: Package, std: Array<string>): Package {
+  // Since alan is blocking the re-asignation of pkg.std we create a new object as a work around
+  return new Package {
+    std: pkg.std + std,
+    dependencies: pkg.dependencies,
+    block: pkg.block,
+    fullBlock: pkg.fullBlock,
+  };
+}
+
+export fn dependency(pkg: Package, url: string): PackageDependency {
+  const emptyBlock = new Array<string> [];
+  const emptyFullBlock = new Array<string> [];
+  // The dependency url have to follow the git URLs standard https://git-scm.com/docs/git-clone#_git_urls
+  // The revisions (branches, tags or SHAs) will be delimited using a # (as npm or other package managers)
+  const parts = url.split('#');
+  const remote = parts[0] || '';
+  const rev = parts[1] || '';
+  const remoteParts = remote.split('/');
+  const gitRepo = remoteParts[remoteParts.length() - 1] || '';
+  const repo = gitRepo.split('.')[0] || '';
+  const possibleSSH = remoteParts[remoteParts.length() - 2] || '';
+  const possibleSSHParts = possibleSSH.split(':');
+  const group = possibleSSHParts[possibleSSHParts.length() - 1] || '';
+  const dep = new Dependency {
+    url: url,
+    block: emptyBlock,
+    fullBlock: emptyFullBlock,
+    group: group,
+    name: repo,
+    rev: rev,
+    remote: remote,
+  };
+  return new PackageDependency {
+    pkg: pkg,
+    dep: dep,
+  };
+}
+
+export fn add(pkgd: PackageDependency): Package {
+  pkgd.pkg.dependencies.push(pkgd.dep);
+  return pkgd.pkg;
+}
+  
+export fn block(pkg: Package, block: string): Package {
+  pkg.block.push(block);
+  return pkg;
+}
+
+export fn block(pkg: Package, block: Array<string>): Package {
+  pkg.block = pkg.block + block;
+  return pkg;
+}
+
+export fn block(pkgd: PackageDependency, block: string): PackageDependency {
+  pkgd.dep.block.push(block);
+  return pkgd;
+}
+
+export fn block(pkgd: PackageDependency, block: Array<string>): PackageDependency {
+  pkgd.dep.block = pkgd.dep.block + block;
+  return pkgd;
+}
+
+export fn fullBlock(pkg: Package, block: string): Package {
+  pkg.fullBlock.push(block);
+  return pkg;
+}
+
+export fn fullBlock(pkg: Package, block: Array<string>): Package {
+  pkg.fullBlock = pkg.fullBlock + block;
+  return pkg;
+}
+
+export fn fullBlock(pkgd: PackageDependency, block: string): PackageDependency {
+  pkgd.dep.fullBlock.push(block);
+  return pkgd;
+}
+
+export fn fullBlock(pkgd: PackageDependency, block: Array<string>): PackageDependency {
+  pkgd.dep.fullBlock = pkgd.dep.fullBlock + block;
+  return pkgd;
+}
```

Comment:
Maybe instead of `block` and `fullBlock` we could do `blockWithMocks` and `block` respectively? It seems more evident with naming like that?

Reply:
it is always with mocks, the difference is if it overrides/force them or not. Another suggestion with this context?

Reply:
ah nvm, hmmm then maybe as-is works well 👍